lychee-lib/src/filter/mod.rs

```diff
@@ -1,72 +1,140 @@
+    ///   - If it's a mail address and it's configured to ignore mail addresses.
+    ///   - If the IP address belongs to a type that is configured to exclude.
+    ///   - If the scheme of URI is not the allowed scheme.
+    /// 2. Decide whether the URI is *presumably included* or *explicitly included*:
+    ///    - When both excludes and includes rules are empty, it's *presumably included* unless
+    ///      it's a known false positive.
+    ///    - When the includes rules matches the URI, it's *explicitly included*.
+    /// 3. When it's a known *false positive* pattern, it's *explicitly excluded*.
+    /// 4. Decide whether the URI is *presumably excluded* or *explicitly excluded*:
+    ///    - When excludes rules is empty, but includes rules doesn't match the URI, it's
+    ///      *presumably excluded*.
+    ///    - When the excludes rules matches the URI, it's *explicitly excluded*.
     pub fn is_excluded(&self, uri: &Uri) -> bool {
-        // Skip mail?
-        if self.excludes.is_mail_excluded() && uri.scheme() == "mailto" {
-            return true;
-        }
-        // Skip specific IP address?
-        if self.excludes.ip(&uri) {
+        // Skip mail address, specific IP, and scheme
+        if self.is_mail_excluded(uri) || self.is_ip_excluded(uri) || self.is_scheme_excluded(uri) {
             return true;
         }
 
         let input = uri.as_str();
-        if self.includes.is_empty() {
-            if self.excludes.is_empty() {
-                // No regex includes/excludes at all?
-                // Not excluded unless it's a known false positive
-                return Excludes::is_false_positive(input);
+
+        if self.is_includes_empty() {
+            if self.is_excludes_empty() {
+                // Both excludes and includes rules are empty:
+                // *Presumably included* unless it's false positive
+                return is_false_positive(input);
             }
-        } else if self.includes.regex(input) {
-            // Included explicitly (Includes take precedence over excludes)
+        } else if self.is_includes_match(input) {
+            // *Explicitly included* (Includes take precedence over excludes)
             return false;
         }
-        // Exclude well-known false-positives.
-        // This is done after checking includes to allow for user-overwrites.
-        if Excludes::is_false_positive(uri.as_str()) {
-            return true;
-        }
-        if self.excludes.is_empty() {
-            if !self.includes.is_empty() {
-                // In case we have includes and no excludes,
-                // skip everything that was not included
-                return true;
-            }
-        } else if self.excludes.regex(input) {
-            // Excluded explicitly
+
+        if is_false_positive(input)
+        // Exclude well-known false-positives
+        // Performed after checking includes to allow user-overwriddes
```

Comment:
overwriddes -> overwrites

Reply:
This is a common misusage.

See the section title: https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html